oxenss/common/serialize.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <string>
+#include <string_view>
+#include <stdint.h>
+
+namespace oxenss {
+enum class Serialise {
+    Read,
+    Write,
+};
```

Comment:
This enum should not exist.

1. It is a bool in disguise.
2. It is only ever used as a run-time parameter for a *known-at-compile-time choice* as whether to make a function do A or B, where A and B are doing fundamentally different (and in fact, opposite) things.

Effectively this just seems to end up being used to hide the word "write" or "read" in a function *argument* rather than in a function *name*.